src/averageAges.js

```diff
@@ -38,6 +50,17 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  let women;
+
+  !withChildren
+    ? women = people.filter(el => el.sex === 'f')
+    : women = people.filter(el =>
+      people.some(child => el.name === child.mother) && (el.sex === 'f'));
+
+  const age = women.map(year => year.died - year.born);
+  const result = (age.reduce((a, b) => a + b) / women.length);
+
+  return result;
```

Comment:
This additional variable is unnecessary here. Return value immediately
```suggestion
  return (age.reduce((a, b) => a + b) / women.length);
```